middleware/TestingHandler.md

```diff
@@ -1,10 +1,16 @@
 # Testing Handler
-A piece of middleware that helps to mock different kinds of responses.
+A piece of middleware that helps to mock different kinds of responses. The goal of this testing handler is to allow a developer to test their application code with random errors injected into their application so that they can see whether the application handles server generated failures. Additionally, a developer can set manual responses to test specific conditions of our client library and the Graph API. 
 
 ## Objectives
-Provide a component, so that application developers can test their application with different kinds of responses (different response codes).
+Provide a configurable testing handler component that, by default, generates random error responses, and can be configured to generate tester specified errors.
 
 ## Requirements
+* MUST support response pass-through for non-intercepted responses.
```

Comment:
@darrelmiller 
Right now whenever we have to use Manual Map, we have to declare testing handler in this way:
```js
const testingHandler = new TestingHandler(new TestingHandlerOptions(TestingStrategy.MANUAL), manualMap);
```

So, now if we need to use Random mode for any of the request, we mention it explicitly in the request, eg:
```js
client
	.api("/me")
	.version("beta")
	.middlewareOptions([new TestingHandlerOptions(TestingStrategy.RANDOM)])
	.get()
	.then((res) => {
		console.log("Found", res, "users");
	})
	.catch((err) => {
		console.log(err);
	});
```
**This is done just to override the default TestingHandlerOptions, which has Manual mode for now.**

Now, as the specs have changed, so if the user does not specify any TestingHandlerOptions at request level, and the request hasn't been mapped in the manual map to any response code, in that case should we pass the request to finalHandler or should we report this as an error situation (as Random mode hasn't been explicitly mentioned).

Reply:
It should randomly return a random error.  e.g. if configured with 10% failure rate, then approximately 1 out of every 10 requests should return a random error.  The rest should be passed through.